source/h5vol/H5VolUtil.c

```diff
@@ -0,0 +1,232 @@
+#include "H5VolUtil.h"
+#include "H5VolError.h"
+
+void *safe_calloc(size_t n, size_t s, unsigned long line)
+{
+    if (n == 0)
+    {
+        return NULL;
+    }
+
+    void *p = calloc(n, s);
+    if (!p)
+    {
+        fprintf(stderr, "[%s:%ld]Out of memory at calloc (%ld, %ld)\n",
+                __FILE__, line, (unsigned long)n, (unsigned long)s);
+        exit(EXIT_FAILURE);
+    }
+    return p;
+}
+
+void *safe_malloc(size_t n, unsigned long line)
+{
+    if (n == 0)
+    {
+        return NULL;
+    }
+
+    void *p = malloc(n);
+    if (!p)
+    {
+        fprintf(stderr, "[%s:%ld]Out of memory at malloc (%ld bytes)\n",
+                __FILE__, line, (unsigned long)n);
+        exit(EXIT_FAILURE);
+    }
+    return p;
+}
+
+void safe_free(void *p)
+{
+    if (p == NULL)
+    {
+        return;
+    }
+    free(p);
+    p = NULL;
```

Comment:
Since `*p` is passed by value then setting it to `NULL` doesn't actually do anything here.  The same effect would be achieved from
```c++
void safe_free(void *p)
{
    if (p)
    {
        free(p);
    }
}
```

Reply:
Random comment: `safe_free` does do exactly the same as `free` -- according to the C standard (at least since 89), `free()` does nothing if `NULL` is passed to it.

Reply:
Thanks for the catch! I fixed it.